ucp/2024-03-planes-apis.md

```diff
@@ -0,0 +1,335 @@
+    is TrackedResourceRequired<AzurePlaneResourceProperties, "System.Planes/azure", "azure"> {
+    @doc("The plane name.")
+    @segment("azure")
+    @path
+    @key("planeName")
+    name: ResourceNameString;
+}
+
+@doc("The Plane properties.")
+model AzurePlaneResourceProperties {
+    @doc("The status of the asynchronous operation.")
+    @visibility("read")
+    provisioningState?: ProvisioningState;
+
+    @doc("The URL used to proxy requests.")
+    url: string;
+}
+```
+
+**Radius Plane**
+
+The new resource type is `System.Planes/radius`. The Radius plane represents a tenant of Radius. We automatically create one called `local` when you initialize Radius.
+
+The only setting we define for the Radius plane is the list of resource providers and their addresses. This will be revisited as part of the user-defined-types.
+
+```typespec
+@doc("The Radius plane resource.")
+model RadiusPlaneResource
+    is TrackedResourceRequired<RadiusPlaneResourceProperties, "System.Planes/radius"> {
+    @doc("The plane name.")
+    @segment("radius")
+    @path
+    @key("planeName")
+    name: ResourceNameString;
+}
+
+@doc("The Plane properties.")
+model RadiusPlaneResourceProperties {
+    @doc("The status of the asynchronous operation.")
+    @visibility("read")
+    provisioningState?: ProvisioningState;
+
+    @doc("Resource Providers for UCP Native Plane")
+    resourceProviders: Record<string>;
+}
+```
+
+## Alternatives considered
+
+The status quo is not good. I'm working on adding the Kubernetes plane, and I didn't want to make this worse. I don't think keep the existing design with a discriminated union is a reasonable idea.
+
+I've coded up a version of this proposal and it simplifies validation in the code that works with planes. This makes the design most consistent with our existing resource providers.
+
+## Test plan
+
+These features are tested by integration tests since they don't need to change any state outside of UCP. I'm expanding the integration tests to cover all of the CRUDL APIs for the new resource types. I'm also improving the consistency of these APIs by including fields like `tags`, `systemData`, and `location` that are present in the API definition but not implemented.
+
+## Security
+
+No changes to security model. 
+
+## Compatibility (optional)
+
+This is a breaking change. Fortunately the only consumer of these APIs is inside UCP so the impact is limited.
+
+UCP initializes the "default planes" as part of its startup sequence, so it should "just work" for an existing Radius installation.
+
+## Monitoring
+
+No new monitoring needed. 
+
+## Development plan
+
+This is an API breaking change so it can't be staged. 
+
+## Open issues
+
+**Naming?**
+
+I'd like some help with naming. I'm not sure that `System.Planes/aws` is right. It feels like we're defining more namespaces than necessary. 
+
+Some other ideas I had:
+
+- `System.Resources/awsPlane`
+- `System.Aws/planes` (we already have `System.Aws/credentials`)
```

Comment:
From discussion in the design meeting we chose `System.Aws/planes` as the style of the API. This is recommended for consistency with `System.Aws/credentials`. This pattern also scales up as we add new things.

I'll update the design doc with this decision.